substrate/frame/scheduler/src/tests.rs

```diff
@@ -1501,8 +1501,11 @@ fn scheduler_handles_periodic_unavailable_preimage() {
 		run_to_block(4);
 		assert_eq!(logger::log().len(), 1);
 
-		// Unnote the preimage
+		// As the public api doesn't support to remove a noted preimage, we need to first unnote it
```

Comment:
@bkchr the admin origin can unnote it here. Which is why i added that `if is_requested...` before.  
Doing this still causes an underflow `assert_ok!(Preimage::unnote_preimage(RuntimeOrigin::root(), hash));`.

Reply:
If the admin origin does this, it is a manual intervention. However, the code does not really underflows as the logic checks that the `count` is bigger than `1` etc. 

I don't want to modify production code to make it work in the edge case for a test.